src/index.rs

```diff
@@ -154,6 +157,40 @@ impl Searcher {
             licenses,
         })
     }
+
+    pub fn completions(&self, term: &str) -> Result<HashSet<String>> {
+        let searcher = self.reader.searcher();
+
+        let de_stem = searcher.index().tokenizers().get("de_stem").unwrap();
+
+        let mut tokens = de_stem.token_stream(term);
+        let mut token = String::new();
+
+        while tokens.advance() {
+            token = take(&mut tokens.token_mut().text);
+        }
+
+        if token.is_empty() {
+            return Ok(HashSet::new());
+        }
+
+        let lower = &format!("{}a", token);
```

Comment:
Instead of manually searching prefixes, we should consider combing all but the last (presumably incomplete) term of the query with the `TermsAggregation` to quickly extracted a weighted set of terms which can be filtered for prefixes.